packages/http-serialization/src/index.spec.ts

```diff
@@ -0,0 +1,338 @@
+} from "./";
+import { requests } from "./requests.fixture";
+import { responses } from "./responses.fixture";
+import { HttpRequest, HttpResponse } from "@aws-sdk/types";
+
+describe("parseRequest", () => {
+  for (const [serialized, expected] of requests) {
+    it(`should parse the request: ${serialized}`, () => {
+      expect(parseRequest(serialized)).toEqual(expected);
+    });
+  }
+
+  it("should reject invalid messages", () => {
+    const serialized = `CONNECT 
+Host: server.example.com:80 
+Proxy-Authorization: basic aGVsbG86d29ybGQ=`;
+
+    expect(() => parseRequest(serialized)).toThrowError(
+      "Invalid request string"
+    );
+  });
+
+  it("should parse multiline headers", () => {
+    const serialized = `GET / HTTP/1.1
+Host:example.amazonaws.com:80
+My-Header1:value1
+  value2
+     value3
+X-Amz-Date:20150830T123600Z`;
+
+    expect(parseRequest(serialized)).toEqual({
+      method: "GET",
+      path: "/",
+      headers: {
+        host: "example.amazonaws.com:80",
+        "my-header1": "value1,value2,value3",
+        "x-amz-date": "20150830T123600Z"
+      },
+      hostname: "example.amazonaws.com",
+      port: 80
+    });
+  });
+
+  it("should parse duplicate headers", () => {
+    const serialized = `GET / HTTP/1.1
+Host:example.amazonaws.com
+My-Header1:value2
+My-Header1:value2
+My-Header1:value1
+X-Amz-Date:20150830T123600Z`;
+
+    expect(parseRequest(serialized)).toEqual({
+      method: "GET",
+      path: "/",
+      headers: {
+        host: "example.amazonaws.com",
+        "my-header1": "value2,value2,value1",
+        "x-amz-date": "20150830T123600Z"
+      },
+      hostname: "example.amazonaws.com"
+    });
+  });
+
+  it("should reject messages with invalid headers", () => {
+    const serialized = `CONNECT server.example.com:80 HTTP/1.1 
+headerValue
+Host: server.example.com:80 
+Proxy-Authorization: basic aGVsbG86d29ybGQ=`;
+
+    expect(() => parseRequest(serialized)).toThrowError(
+      "Invalid line encountered: headerValue"
+    );
+  });
+
+  it("should parse absolute URL targets", () => {
+    expect(
+      parseRequest(
+        "GET https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/GET HTTP/1.1"
+      )
+    ).toEqual({
+      method: "GET",
+      protocol: "https:",
+      hostname: "developer.mozilla.org",
+      path: "/en-US/docs/Web/HTTP/Methods/GET",
+      headers: {}
+    });
+  });
+});
+
+describe("parseResponse", () => {
+  for (const [serialized, expected] of responses) {
+    it(`should parse the response: ${serialized}`, () => {
+      expect(parseResponse(serialized)).toEqual(expected);
+    });
+  }
+
+  it("should reject invalid messages", () => {
+    const serialized = `SFTP/1.1 204 No Content`;
+
+    expect(() => parseResponse(serialized)).toThrowError(
+      "Invalid response string"
+    );
+  });
+});
+
+describe("serializeRequest", () => {
+  it("should serialize requests with string bodies", () => {
+    const request: HttpRequest<string> = {
```

Comment:
To fix CI this change is needed here, and other places
```suggestion
    const request = new HttpRequest({
```

Reply:
Will fix it. Interestingly it wasn't caught on my local

Reply:
Resolved in https://github.com/aws/aws-sdk-js-v3/pull/712/commits/9b81844569f2d818feb33458ba8c39546805153e